yoda/src/gen_data.py

```diff
@@ -0,0 +1,176 @@
+import argparse
+import logging
+import os
+import random
+
+import jsonlines
+import tqdm
+import yaml
+from transformers import AutoTokenizer
+
+from utils import qa_processing, data_reader
+
+from dotenv import load_dotenv
+load_dotenv('.env')
+
+# set the log format to stdout
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(levelname)s - %(message)s')
+
+
+def parse_args() -> argparse.Namespace:
+    """
+    Parse command line arguments.
+
+    Returns:
+        argparse.Namespace: Parsed command line arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config_file", type=str, default="configs/sn_expert_conf.yaml",
+                        help="path to the config file")
+    parser.add_argument("--purpose", type=str, default="pretrain",
+                        choices=["pretrain", "finetune", "both"],
+                        help="purpose of the training data, pretrain or finetune or both")
+
+    return parser.parse_args()
+
+
+# create training data for pretraining
+
+
+def pretrain(articles, data_dir: str, processed_data_folder: str = 'processed_data',
+             output_file_name: str = 'article_data.jsonl') -> list:
+    article_data = []
+    for d in articles:
+        article = d["article"]
+        # clean the article
+        article = article.replace(".\n", ".[NEWLINE]]")
+        article = article.replace("\n", "")
+        article = article.replace("[NEWLINE]]", "\n")
+        article_data.append({"prompt": "", "completion": article})
+        # print sample data for debugging
+        # print(article)
+
+    output_path = os.path.join(
+        data_dir, processed_data_folder, output_file_name)
+    with jsonlines.open(output_path, "w") as wf:
+        wf.write_all(article_data)
+
+    return article_data
+
+
+# create training data for finetuning
+def finetune(articles, data_dir, tokenizer, pretrain_data=None) -> None:
+    QA_GEN_TEMPLATE = """
```

Comment:
same as before, if we can pull out all the prompts in a single place (config file)?